### PR TITLE
Refactors create thread reaction command

### PR DESCRIPTION
**Pull request by Rotorsoft**
## Link to Issue
Closes: #8934

## Description of Changes
- See ticket

## Test Plan
- Upvoting on threads should work as expected

### CODE REVIEW

**Review by rbennettcw — APPROVED** [review]
tested locally, seems to work

File: libs/model/src/services/stakeHelper.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  const stakeBalances = await contractHelpers.getNamespaceBalance(
+    community.namespace_address!,
+    stake.stake_id,
+    node.eth_chain_id,
+    [address],
+  );
+  const stakeBalance = stakeBalances[address];
+  if (BigNumber.from(stakeBalance).lte(0))
+    throw new InvalidState('Must have stake to upvote');
```
[the remark this answers is not in the document]
Comment (Rotorsoft) [reply libs/model/src/services/stakeHelper.ts:43]: agree... we can refine this in the future to remove "guard like" behavior

File: libs/model/src/thread/CreateThreadReaction.command.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+              reaction: payload.reaction,
+              calculated_voting_weight,
+              canvas_hash: payload.canvas_hash,
+              canvas_signed_data: payload.canvas_signed_data,
+            },
+            transaction,
+          });
+
+          address.last_active = new Date();
+          await address.save({ transaction });
```
[the remark this answers is not in the document]
Comment (Rotorsoft) [reply libs/model/src/thread/CreateThreadReaction.command.ts:67]: we can have it in a policy for sure, but I'm keeping it in the mutation assuming we wanted atomic consistency... but eventual consistency should be fine too. In any case, the performance hit is minimal

File: libs/model/src/thread/CreateThreadReaction.command.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    ],
+    body: async ({ actor, payload }) => {
+      const thread = await models.Thread.findOne({
+        where: { id: payload.thread_id },
+      });
+      if (!mustExist('Thread', thread)) return;
+      if (thread.archived_at)
+        throw new InvalidState(CreateThreadReactionErrors.ThreadArchived);
+
+      const address = await models.Address.findOne({
```
Comment (kurtisassad) [inline libs/model/src/thread/CreateThreadReaction.command.ts:32]: This address check on the community with the userId passed in the JWT seems like a pretty common thing we do, shouldn't it be abstracted into a middleware function?

Comment (Rotorsoft) [reply libs/model/src/thread/CreateThreadReaction.command.ts:32]: yep, I'm building the pattern inside authorization middleware... will need refactoring soon after we finish the first batch of commands ... checkout the next command in the pipeline for more changes around this area

**Review by kurtisassad — APPROVED** [review]

**Pull request merged**